commands/utils/params.go

```diff
@@ -208,14 +208,27 @@ func (g *Git) setDefaultsIfNeeded(gitParamsFromEnv *Git) (err error) {
 		}
 		g.RepoName = gitParamsFromEnv.RepoName
 	}
-	if len(g.Branches) == 0 && len(gitParamsFromEnv.Branches) == 0 {
-		var branch string
-		if branch, err = GetBranchFromDotGit(); err != nil {
-			return
+	if g.EmailAuthor == "" {
+		if g.EmailAuthor = getTrimmedEnv(GitEmailAuthorEnv); g.EmailAuthor == "" {
+			g.EmailAuthor = frogbotAuthorEmail
+		}
+	}
+	// When pull request ID is provided, no need to continue and extract unrelated env params.
+	if g.PullRequestDetails.ID != 0 || gitParamsFromEnv.PullRequestDetails.ID != 0 {
```

Comment:
why we need the second condition here?

Reply:
isPullRequestContext := g.PullRequestDetails.ID != 0
if isPullRequestContext {
return
 }